pyqtgraph/graphicsItems/PlotCurveItem.py

```diff
@@ -514,18 +536,18 @@ def updateData(self, *args, **kargs):
             self.opts['connect'] = kargs['connect']
         if 'pen' in kargs:
             self.setPen(kargs['pen'])
-        if 'shadowPen' in kargs and kargs['shadowPen'] is not None:
+        if 'shadowPen' in kargs:
             self.setShadowPen(kargs['shadowPen'])
-        if 'fillLevel' in kargs and kargs['fillLevel'] is not None:
+        if 'fillLevel' in kargs:
             self.setFillLevel(kargs['fillLevel'])
         if 'fillOutline' in kargs:
             self.opts['fillOutline'] = kargs['fillOutline']
-        if 'brush' in kargs and kargs['brush'] is not None:
+        if 'brush' in kargs:
             self.setBrush(kargs['brush'])
         if 'antialias' in kargs:
             self.opts['antialias'] = kargs['antialias']
-
-        self.opts['skipFiniteCheck'] = kargs.get('skipFiniteCheck', False)
+        if 'skipFiniteCheck' in kargs:
+            self.opts['skipFiniteCheck'] = kargs['skipFiniteCheck']
```

Comment:
This updates skipFiniteCheck only on request. A default False value will be used if it is never explicitly set.